testsuite/MDAnalysisTests/converters/test_rdkit.py

```diff
@@ -484,82 +498,188 @@ def test_set_atom_property(self, attr, value, getter):
+        ("smi", "c1c2c(=O)n3cccc(C)c3nc2n(C)c(=N)c1C(=O)NCc1cnccc1"),
+        ("smi", "N#Cc1c[nH]c(C(=O)NC(=O)c2cc[n+]([O-])cc2)n1"),
+        ("smi", "C[C@@H](Oc1cc(F)ccc1Nc1ncnc2cc(N=S3(=O)CCC3)cc(F)c12)C(=O)NCC#N"),
+        ("smi", "[O-][n+]1onc2ccccc21"),
+        ("smi", "Cc1cc[n+](CC[n+]2ccc(C)cc2)cc1"),
+        ("smi", "[O-]c1ccccc1"),
+        ("smi", "[O-]C=CC=CCC=CC=[N+](C)C"),
+        # test amino acids
+        ("aa", "A"),
+        ("aa", "G"),
+        ("aa", "I"),
+        ("aa", "L"),
+        ("aa", "P"),
+        ("aa", "V"),
+        ("aa", "F"),
+        ("aa", "W"),
+        ("aa", "Y"),
+        ("aa", "D"),
+        ("aa", "E"),
+        ("aa", "R"),
+        ("aa", "H"),
+        ("aa", "K"),
+        ("aa", "S"),
+        ("aa", "T"),
+        ("aa", "C"),
+        ("aa", "M"),
+        ("aa", "N"),
+        ("aa", "Q"),
+        # test nucleic acids
+        ("na", "A"),
+        ("na", "T"),
+        ("na", "U"),
+        ("na", "C"),
+        ("na", "G"),
+    ])
+    def test_order_independant(self, input_type, input_str):
         # generate mol with hydrogens but without bond orders
-        ref = Chem.MolFromSmiles(smi_in)
-        template = Chem.AddHs(ref)
-        for atom in template.GetAtoms():
-            atom.SetIsAromatic(False)
-            atom.SetFormalCharge(0)
-        for bond in template.GetBonds():
-            bond.SetIsAromatic(False)
-            bond.SetBondType(Chem.BondType.SINGLE)
-
+        if input_type == "smi":
+            ref = Chem.MolFromSmiles(input_str)
+        elif input_type == "na":
+            ref = Chem.MolFromSequence(input_str, flavor=9)
+        else:
+            ref = Chem.MolFromSequence(input_str)
+        template = self.mol_to_template(ref)
         # go through each possible starting atom
         for a in template.GetAtoms():
             smi = Chem.MolToSmiles(template, rootedAtAtom=a.GetIdx())
             m = Chem.MolFromSmiles(smi, sanitize=False)
             for atom in m.GetAtoms():
                 atom.SetFormalCharge(0)
+                atom.SetIsAromatic(False)
                 atom.SetNoImplicit(True)
             m.UpdatePropertyCache(strict=False)
             _infer_bo_and_charges(m)
             m = _standardize_patterns(m)
             Chem.SanitizeMol(m)
             m = Chem.RemoveHs(m)
-            assert m.HasSubstructMatch(ref) and ref.HasSubstructMatch(
-                m), (f"(input) {Chem.MolToSmiles(ref)} != "
-                     f"{Chem.MolToSmiles(m)} (output) root atom {a.GetIdx()}")
+            match = m.HasSubstructMatch(ref) and ref.HasSubstructMatch(m)
+            if not match:
+                # try resonance structures for charged conjugated systems
+                for mol in Chem.ResonanceMolSupplier(m, maxStructs=20):
+                    match = (mol.HasSubstructMatch(ref) and
+                             ref.HasSubstructMatch(mol))
+                    if match:
+                        break
```

Comment:
Would it be difficult to break out the test cases where you run into resonance problems, and explicitly test that resonance structure, in a separate test? i.e. `def test_order_independent_resonance`

Reply:
I think this would mostly concern the molecules with a charge next to a conjugated system, which is a good portion of the tests here, but I'm not sure how separating the tests is going to make things more readable. It would IMO make things more difficult when/if we later add other tests.
I think a good option is to reorganize the test code with more functions with descriptive names, as you suggested in other comments